### ISSUE TITLE
dateInput lacks semantic accessibility

### ISSUE THREAD

**Issue by jooyoungseo** [rstudio/shiny#2951]
### System details

* Browser Version: any

* Session info:

``` r
xfun::session_info("shiny")
#> R version 4.0.2 (2020-06-22)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19041)
#> 
#> Locale:
#>   LC_COLLATE=English_United States.1252 
#>   LC_CTYPE=English_United States.1252   
#>   LC_MONETARY=English_United States.1252
#>   LC_NUMERIC=C                          
#>   LC_TIME=English_United States.1252    
#> system code page: 949
#> 
#> Package version:
#>   base64enc_0.1.3   BH_1.72.0.3       commonmark_1.7    crayon_1.3.4     
#>   digest_0.6.25     fastmap_1.0.1     glue_1.4.1        graphics_4.0.2   
#>   grDevices_4.0.2   htmltools_0.5.0   httpuv_1.5.4      jsonlite_1.7.0   
#>   later_1.1.0.1     magrittr_1.5      methods_4.0.2     mime_0.9         
#>   promises_1.1.1    R6_2.4.1          Rcpp_1.0.4.6      rlang_0.4.6      
#>   shiny_1.5.0       sourcetools_0.1.7 stats_4.0.2       tools_4.0.2      
#>   utils_4.0.2       withr_2.2.0       xtable_1.8.4
```

<sup>Created on 2020-06-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### Example application

``` r
library(shiny)

example(dateInput)
```

### Describe the problem in detail

1. Each input label element is annotated with `for=` attribute but refers to the `<div>` that contains both the label and the following textbox, which means the textbox itself has no accessible label. So that's messed up for assistive technology users.
2. In the pop-up calendar grid, the three buttons (previous, next, and datepicker switch) are completely unannotated and inaccessible for screen readers.
3. No semantic hint is presented to screen readers when the calendar popup appears and disappears.

Greatly assisted by @@gtritchie in identifying those issues.

**Comment by jooyoungseo** [comment]
As a solution, we can employ [accessible and bootstrap compatible datepicker](https://github.com/eureka2/ab-datepicker).
